packages/vite/src/node/plugins/esbuild.ts

```diff
@@ -432,32 +433,51 @@ function prettifyMessage(m: Message, code: string): string {
   return res + `\n`
 }
 
-const tsconfckParseOptions: TSConfckParseOptions = {
-  cache: new Map<string, TSConfckParseResult>(),
-  tsConfigPaths: undefined,
-  root: undefined,
-  resolveWithEmptyIfConfigNotFound: true,
+let tsconfckRoot: string | undefined
+let tsconfckParseOptions: TSConfckParseOptions | Promise<TSConfckParseOptions> =
+  { resolveWithEmptyIfConfigNotFound: true }
```

Comment:
Shouldn't we have these cache as `WeakMap<ResolvedConfig,...>`? It would be good to make it safe to create two esbuild plugins programmatically.

Reply:
`transformWithEsbuild` mess it all up 😄 It doesn't have a reference to `ResolvedConfig`, and we can't get the "first value" of the weak map because the API doesn't support that.

I refactored this part a few times and decided to keep most of it as is at the end. Maybe we can hae `transformWithEsbuild` accept a `ResolvedConfig` too in the future.